cloudant-client/src/main/java/com/cloudant/client/api/ClientBuilder.java

```diff
@@ -607,4 +610,62 @@ public ClientBuilder readTimeout(long readTimeout, TimeUnit readTimeoutUnit) {
         return this;
     }
 
+    /**
+     * Sets Cloudant client credentials by inspecting the VCAP_SERVICES Cloud Foundry environment
+     * variable. The VCAP_SERVICES variable contains connection information to access a Bluemix
+     * service instance.
+     *
+     * @return a new ClientBuilder for the account
+     */
+    public static ClientBuilder bluemix() {
+        return ClientBuilder.bluemix(null);
+    }
+
+    /**
+     * Sets Cloudant client credentials by inspecting the VCAP_SERVICES Cloud Foundry environment
+     * variable. The VCAP_SERVICES variable contains connection information to access a Bluemix
+     * service instance.
+     * <P>
+     * Note: Specifying a service name is only required when multiple Cloudant services are
+     * available.
+     * </P>
+     *
+     * @param serviceName name of Bluemix service instance
+     * @return a new ClientBuilder for the account
+     */
+    public static ClientBuilder bluemix(String serviceName) {
+        Gson gson = new GsonBuilder().create();
+
+        String vcap_env = System.getenv("VCAP_SERVICES");
+        if (vcap_env == null) {
+            throw new IllegalArgumentException("VCAP_SERVICES is not present.");
```

Comment:
I'm in two minds about these first three `IllegalArgumentException`s. The javadoc for `IllegalArgumentException` says:
>Thrown to indicate that a method has been passed an illegal or inappropriate argument.

It seems slightly weird to throw it when the argument is correct, but the environment is wrong. That said it could be argued that the method argument is inappropriate when the environment is not correct. However, in my mind `IllegalArgumentException` says I can change the argument and not get the exception, but in this case (and the case of invalid JSON or no Cloudant services bound) that wouldn't be the case.

I'd appreciate other's thoughts on this, but perhaps these first three should really be an `IllegalStateException`.

I'm happy with the last two being `IllegalArgumentException` because changing the argument used could resolve the exception.